sunpy/map/tests/test_mapbase.py

```diff
@@ -1546,6 +1546,14 @@ def test_rotation_rect_pixelated_data(aia171_test_map):
     rect_rot_map = rect_map.rotate(30 * u.deg)
     rect_rot_map.peek()
 
+@figure_test
+def test_draw_contours_with_transform(aia171_test_map):
+    aia_map = sunpy.map.Map(aia171_test_map)
+    fig = plt.figure(figsize=(8, 4))
+    ax = fig.add_subplot(projection=aia_map)
+    aia_map.plot(axes=ax, clip_interval=(1, 99.99)*u.percent)
+    aia_map.draw_contours([-2,2]*u.percent, transform=ax.get_transform('pixel'))
```

Comment:
Can you try to replace the transfrom with something like https://matplotlib.org/stable/api/transformations.html#matplotlib.transforms.Affine2D

We want to make sure the transform works and picking one that will produce bad looking contours should make the figure test more useful.